index.js

```diff
@@ -1,26 +1,30 @@
 const core = require('@actions/core');
-
-const { execSync } = require('child_process');
+const io = require('@actions/io');
+const exec = require('@actions/exec');
 
 const host = core.getInput('host');
 const port = core.getInput('port');
 const socketPath = core.getInput('socket-path');
 const key = core.getInput('key');
+const lifetimeInSeconds = core.getInput('lifetime_in_seconds');
 
-console.log(`Attempting to create ${socketPath}...`);
-try {
-    execSync(
-        'mkdir -p ~/.ssh && ' +
-        `ssh-keyscan${port ? ` -p ${port}` : ''} "${host}" >> ~/.ssh/known_hosts && ` +
-        `eval $(ssh-agent -a "${socketPath}") && ` +
-        `echo "${key}" | base64 -d | ssh-add -`
-    );
-} catch (e) {
-    console.error(e.message);
-    process.exit(1);
-}
+async function run() {
+    console.log('checking if ~/.ssh is created...');
+    await io.mkdirP('~/.ssh');
+
+    console.log('ssh-keyscan the provided domain...');
+    let portCommand = port ? `-p ${port}` : '';
+    await exec.exec('ssh-key', [portCommand, host, '>> ~/.ssh/known_hosts']);
```

Comment:
I think typo: I think `ssh-key` is not a thing (probably meant `ssh-keyscan`). We should also probably use `ssh-keygen -R $host -f ~/.ssh/known_hosts` to delete before adding another

Reply:
thanks added this